runtime-modules/forum/src/lib.rs

```diff
@@ -787,10 +783,20 @@ decl_module! {
             });
 
             // Reserve cleanup pay off in the thread account
-            Self::transfer_to_state_cleanup_treasury_account(cleanup_pay_off, new_thread_id, &account_id);
+            Self::transfer_to_state_cleanup_treasury_account(
```

Comment:
In case of error here - we don't have "mutation safety". Please, consider moving it before the infallible operations (like `<ThreadById<T>>::mutate`)